.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,156 @@
+name: integration-tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+    paths-ignore:
+      [
+        "abstractions/**",
+        "authentication/**",
+        "serialization/**",
+        "http/**",
+        "**.md",
+        ".vscode/**",
+        "**.svg",
+      ]
+  pull_request:
+    paths-ignore:
+      [
+        "abstractions/**",
+        "authentication/**",
+        "serialization/**",
+        "http/**",
+        "**.md",
+        ".vscode/**",
+        "**.svg",
+      ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 7.0.x
+      - name: Restore workloads
+        run: dotnet workload restore
+      - name: Restore dependencies
+        run: dotnet restore kiota.sln
+      - name: Build
+        run: dotnet publish ./src/kiota/kiota.csproj -c Release -p:PublishSingleFile=true -p:PublishReadyToRun=true -o ./publish
+        # -p:PublishTrimmed=true -p:PublishAot=true should be enabled to make test run faster, but there are still limitations
+      - uses: actions/upload-artifact@v3
+        with:
+          name: generator
+          path: publish
+      - uses: actions/upload-artifact@v3
+        with:
+          name: it
+          path: it
+      - uses: actions/upload-artifact@v3
+        with:
+          name: tests
+          path: tests/**/*.yaml
+  java-it:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: generator
+          path: publish
+      - uses: actions/download-artifact@v3
+        with:
+          name: it
+          path: it
+      - uses: actions/download-artifact@v3
+        with:
+          name: tests
+          path: tests
+      - run: chmod +x ./publish/kiota
+      - name: Generate Java Code
+        run: |
+          ./publish/kiota generate --language Java --openapi ./tests/Kiota.Builder.IntegrationTests/InheritingErrors.yaml --output ./it/java/src
```

Comment:
Super cool to see this shaping up, excellent work @baywet !

Do you think that we can run the `generate` commands on a matrix of languages and share basically all the "setup" phase?

Reply:
I thought of that but the build steps will be different afterwards, I might be able to fix that by adding conditions, still exploring at this point.

Reply:
I was thinking the same, one option is to produce an artifact with all the generated sources and use separate `job`s to validate the specific languages.